pkg/agent/proxy/types/groupcounter.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"sync"
+
+	binding "github.com/vmware-tanzu/antrea/pkg/ovs/openflow"
+	k8sproxy "github.com/vmware-tanzu/antrea/third_party/proxy"
+)
+
+// GroupCounter generates and manages global unique group ID.
+type GroupCounter interface {
+	// Get generates a global unique group ID for a specific service.
+	// If the group ID of the service has been generated, then return the
+	// prior one. The bool return value indicates whether the groupID is newly
+	// generated.
+	Get(svcPortName k8sproxy.ServicePortName) (binding.GroupIDType, bool)
+	// Recycle removes a Service Group ID mapping. The recycled groupID should
```

Comment:
s/should be able to be reused/can be reused

Reply:
Done.